File: java/json-transform/src/main/java/co/nlighten/jsontransform/functions/TransformerFunctionJsonParse.java
package co.nlighten.jsontransform.functions;

import co.nlighten.jsontransform.functions.common.FunctionContext;
import co.nlighten.jsontransform.functions.common.TransformerFunction;

public class TransformerFunctionJsonParse extends TransformerFunction {
    public TransformerFunctionJsonParse() {
        super();
    }
    @Override
    public CompletionStage<Object> apply(FunctionContext context) {
        var str = context.getString(null);
        if (str == null) {
            return null;
        }
        return context.getAdapter().parse(str);
    }
}

File: javascript/json-transform/src/functions/TransformerFunctionJsonParse.ts
import TransformerFunction from "./common/TransformerFunction";
import FunctionContext from "./common/FunctionContext";
import { JSONBig } from "./common/FunctionHelpers";

class TransformerFunctionJsonParse extends TransformerFunction {
  constructor() {
    super({});
  }

  override async apply(context: FunctionContext): Promise<any> {
    const str = await context.getString(null);
    if (str == null) {
      return null;
    }
    return JSONBig.parse(str);
  }
}

export default TransformerFunctionJsonParse;
